### DISCUSSION TITLE
Installation issue on minikube

### DISCUSSION THREAD

**Discussion by puru786** [kubev2v/forklift#1187] [Q&A]
Hello, I am trying to deploy Forklift on my Minikube setup. However, when I try to install the Forklift controller, it stays stuck in container creating mode. 

I am following this instruction to deploy on Minikube: https://github.com/kubev2v/forklift/blob/main/operator/docs/k8s.md

![Screenshot 2024-11-14 at 5 28 25 PM](https://github.com/user-attachments/assets/d55db17f-1fb2-4f57-8a0c-25c240086458)
But as you see there are few pods stuck in image pull back

Then I deployed the Konveyor-forklift manifest: (https://github.com/kubev2v/forklift/blob/main/operator/forklift-k8s.yaml)

![Screenshot 2024-11-14 at 5 30 37 PM](https://github.com/user-attachments/assets/5658c1c6-577a-418e-ab03-d5c7bce30a92)

Then I deployed the Forklift-controller:

$ cat << EOF | kubectl -n konveyor-forklift apply -f -
apiVersion: forklift.konveyor.io/v1beta1
kind: ForkliftController
metadata:
  name: forklift-controller
  namespace: konveyor-forklift
spec: {}
EOF

but as you can see the controller is stuck in container creating state.
can you please help me figure this issue out?

**Comment by mrnold** [comment]
I tried this out and saw the same thing. I guess the OLM addon for minikube doesn't really work because the image hash has changed, for the time being you can do `kubectl edit catalogsource -n olm operatorhubio-catalog` and change the `e08a1cd21fe72dd1be92be738b4bf1515298206dac5479c17a4b3ed119e30bd4` images to `b43aa26d13ca4ebc77133d0466eb28a53d3cbee2121bd51c93f35e175e3128d1` instead. That got my operatorhubio pods running, at least. I'm sure it will change again at some point, but that hash came from here: https://quay.io/repository/operatorhubio/catalog?tab=tags

I also saw my forklift-controller pod get stuck in ContainerCreating, I think because of this error from the forklift-operator pod: `Failed to find exact match for cert-manager.io/v1.Issuer`

If you are getting that error, you can install cert-manager manually with `kubectl create -f https://operatorhub.io/install/cert-manager.yaml`, then watch for it to finish with `kubectl get csv -n operators` (from "Install" on https://operatorhub.io/operator/cert-manager). That should install `cert-manager.io/v1.Issuer` and the next reconcile loop in forklift-operator should get un-stuck.

This got my forklift-controller pod to go running, although I haven't tried creating providers or plans or anything yet.

**Comment by puru786** [comment]
Hello @mrnold,

Thank you for the help with the installation. I tried to change the image hash but for some reason it did not work. So I had to look for a  work around. I am going to share the workaround in a different comment. But I need your help with one more thing. The OLM and the Forklift components are all installed but Forklift UI and the Forklift Must Gather component are no where to be seen. As per the documentation, the Forklift operator should install these but it didn't for me. It would be of great help if you can guide me in the right direction. Here are the OLM and Forklift pods:

![Screenshot 2024-11-20 at 3 15 22 PM](https://github.com/user-attachments/assets/db513b0d-0693-468c-a932-90f74a1df660)

![Screenshot 2024-11-20 at 3 15 56 PM](https://github.com/user-attachments/assets/bc8db7c0-3e48-4ccd-83f8-f0de72efd3b4)

**Comment by puru786** [comment]
Here are the steps for the installation, in case any one comes looking for it :)

First I downloaded the OLM crd.yaml and olm.yaml file from here: 
https://github.com/operator-framework/operator-lifecycle-manager/releases

Second step is to deploy these files but never do:  
kubectl apply -f crd.yaml 
kubectl apply -f olm.yaml. 

If you use the apply method kubectl will throw an error for the memory size of the CRD too large for it to handle. You have to use the create method:

kubectl create -f crds.yaml
kubectl create -f olm.yaml

Once these are deployed. The next step is to deploy the cert-manager using this: 
kubectl create -f https://operatorhub.io/install/cert-manager.yaml

Then you deploy the forklift manifest:
kubectl apply -f https://raw.githubusercontent.com/kubev2v/forklift/refs/heads/main/operator/forklift-k8s.yaml

Last but not the least is to deploy the controller. 

$ cat << EOF | kubectl -n konveyor-forklift apply -f -
apiVersion: forklift.konveyor.io/v1beta1
kind: ForkliftController
metadata:
  name: forklift-controller
  namespace: konveyor-forklift
spec: {}
EOF


Once you are done, please check the olm and konveyor namespaces and confirm if all the pods are in running state. And you should be good to go from here.

P.S: This installation is strictly for Minikube. I haven't tested this method on a K8s cluster but it should work there as well imo
